extensions/typefully/src/upcoming-scheduled-posts.tsx

```diff
@@ -0,0 +1,63 @@
+import got from "got";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { getPreferenceValues, MenuBarExtra, open } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import { Draft } from "./interfaces/draft";
+import { Preferences } from "./interfaces/preferences";
+
+dayjs.extend(relativeTime);
+
+interface ScheduledPostItemProps {
+  draft: Draft;
+}
+
+const ScheduledPostItem = (props: ScheduledPostItemProps) => {
+  const maxTitleLength = 24;
+  const timeUntilPost = dayjs().to(dayjs(props.draft.scheduled_date));
+
+  return (
+    <MenuBarExtra.Item
+      key={props.draft.id}
+      title={`${props.draft.text_first_tweet.slice(0, maxTitleLength - timeUntilPost.length)}... (${timeUntilPost})`}
+      onAction={() => open(`https://typefully.com/?d=${props.draft.id}`)}
+    />
+  );
+};
+
+export default function Command() {
+  const preferences = getPreferenceValues<Preferences>();
+  const { isLoading, data } = useCachedPromise(
```

Comment:
FYI you could also use `useFetch`, though your implementation is equally good.

Reply:
I saw that - I'm writing a couple of new features at the moment, and the move to the useFetch hook will come with those updates. 🙌